src/common/Utilities/SmartEnum.h

```diff
@@ -23,7 +23,9 @@
 
 struct EnumText
 {
-    EnumText(char const* c, char const* t, char const* d) : Constant(c), Title(t), Description(d) {}
+    EnumText(uint32 i, char const* c, char const* t, char const* d) : Index(i), Constant(c), Title(t), Description(d) { }
```

Comment:
I'm not quite happy with `EnumText` and `ToString` anymore if we do it like this.

@Treeston suggestions?

Reply:
no need to add it to enumtext

literally just make it the opposite of what FromIndex already is